File: epi_judge_python/primitive_divide.py
from test_framework import generic_test


def divide(x: int, y: int) -> int:
    # TODO - you fill in here.
    if x < y:
        return 0
    q = 1
    den = y
    while y <= x:
        y <<= 1
        q <<= 1
    return (q >> 1) + divide(x - (y >> 1), den)


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('primitive_divide.py',
                                       'primitive_divide.tsv', divide))

File: epi_judge_cpp/primitive_divide.cc
#include "test_framework/generic_test.h"
int Divide(int x, int y) {
  // TODO - you fill in here.
  // return 0;

  int result{0};
  while (x >= y) {
    for (auto i = 0; i < sizeof(int) * 8; ++i) {
      if (((y << i) > x) || ((y << i) < y)) {
        result += 1 << (i - 1);
        x -= (y << (i - 1));
        break;
      } 
    }
  }
  return result;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"x", "y"};
  return GenericTestMain(args, "primitive_divide.cc", "primitive_divide.tsv",
                         &Divide, DefaultComparator{}, param_names);
}
